src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/ExpressionLambdaRewriter.cs

```diff
@@ -285,7 +285,7 @@ private BoundExpression VisitArrayAccess(BoundArrayAccess node)
             {
                 var arg = node.Indices[0];
                 var index = Visit(arg);
-                if (node.Type != _int32Type)
+                if (index.Type != _int32Type)
```

Comment:
Consider removing the `if` condition instead, especially since the expression was always true previously.